nntrainer/models/neuralnet.cpp

```diff
@@ -603,6 +603,44 @@ int NeuralNetwork::extendGraph(GraphType graph, std::string prefix) {
   return ML_ERROR_NONE;
 }
 
+std::vector<std::shared_ptr<Layer>>
+NeuralNetwork::getGraph(std::string input_layer, std::string output_layer) {
+  /** count layers after output layer */
+  unsigned int num_layers_remove_end = 0;
+  if (!output_layer.empty()) {
+    for (auto iter = layers.rbegin(); iter != layers.rend(); iter++) {
+      std::cout << (*iter)->getName() << std::endl;
+      if ((*iter)->getName() != output_layer)
+        num_layers_remove_end++;
+      else
+        break;
+    }
+  }
+
+  if (num_layers_remove_end == layers.size())
+    return {};
+
+  /** count layers before input layer */
+  unsigned int num_layers_remove_start = 0;
+  if (!input_layer.empty()) {
+    for (auto iter = layers.begin();
+         iter != layers.end() - num_layers_remove_end; iter++) {
+      std::cout << (*iter)->getName() << std::endl;
+      if ((*iter)->getName() != input_layer)
+        num_layers_remove_start++;
+      else
+        break;
+    }
+  }
+
+  /** copy the graph and return */
+  GraphType ret;
```

Comment:
Maybe later, we might have to use flatgraph type to iterate through the graph or graph itself supports iterators

Reply:
Yes, this will directly depend on graph_core API exposed by the graph.
For now, this is tightly dependent on the graph being implemented as a vector.